src/interface/blas1_interface.hpp

```diff
@@ -811,14 +811,41 @@ typename sb_handle_t::event_t _rotmg(
   auto d1_view = make_vector_view(_d1, inc, vector_size);
   auto d2_view = make_vector_view(_d2, inc, vector_size);
   auto x1_view = make_vector_view(_x1, inc, vector_size);
-  auto y1_view = make_vector_view(_y1, inc, vector_size);
   auto param_view = make_vector_view(_param, inc, param_size);
 
-  auto operation =
-      Rotmg<decltype(d1_view)>(d1_view, d2_view, x1_view, y1_view, param_view);
-  auto ret = sb_handle.execute(operation, _dependencies);
+  if constexpr (std::is_arithmetic_v<container_3_t>) {
+    constexpr helper::AllocType mem_type = std::is_pointer_v<container_0_t>
+                                               ? helper::AllocType::usm
+                                               : helper::AllocType::buffer;
+    auto _y1_tmp = blas::helper::allocate<mem_type, container_3_t>(
+        1, sb_handle.get_queue());
 
-  return ret;
+    auto copy_y1 =
+        blas::helper::copy_to_device(sb_handle.get_queue(), &_y1, _y1_tmp, 1);
```

Comment:
Does copying this need to wait on `_dependencies`?

Reply:
Thank you for your review @hjabird 
I thought it was more asynchronous, but maybe `y1` could be the result of another operation and the copy needs to wait on the dependencies. I will update it.

Reply:
I addressed in [81f5ee7](https://github.com/codeplaysoftware/portBLAS/pull/532/commits/81f5ee7996ac41220e5dd9e794dfabd3062ff4d5). I changed other dependencies accordingly.